crates/spark/src/services/timelock_manager.rs

```diff
@@ -148,11 +148,24 @@ impl TimelockManager {
         &self,
         renew_type: RenewType,
         node: &TreeNode,
-        parent_node: &TreeNode,
+        parent_node: Option<&TreeNode>,
```

Comment:
What do you think about putting the parent node inside the RenewType enum?

```
enum RenewType {
    Node { parent: TreeNode },
    Refund  { parent: TreeNode },
    ZeroTimelock,
}
```

Reply:
Good idea! Done